test/NuGet.Core.Tests/NuGet.Protocol.Tests/Plugins/PluginDiscovererTests.cs

```diff
@@ -519,6 +519,61 @@ public void GetPluginsInNuGetPluginPaths_NoEnvironmentVariables_ReturnsNoPlugins
             Assert.Empty(plugins);
         }
 
+#if !IS_DESKTOP
+        [PlatformFact(Platform.Windows)]
+        public void GetPluginsInNuGetPluginPaths_Windows_CaseInsensitiveMatching()
+        {
+            // Arrange
+            using var testDirectory = TestDirectory.Create();
+
+            var pluginFilePath1 = Path.Combine(testDirectory.Path, "nuget-plugin-test.exe");
+            var pluginFilePath2 = Path.Combine(testDirectory.Path, "NUGET-PLUGIN-TEST.exe"); // Different casing
+
+            File.Create(pluginFilePath1).Dispose();
+            File.Create(pluginFilePath2).Dispose();
+
+            var environmentalVariableReader = new Mock<IEnvironmentVariableReader>();
+            environmentalVariableReader.Setup(env => env.GetEnvironmentVariable(EnvironmentVariableConstants.PluginPaths)).Returns($"{pluginFilePath1}{Path.PathSeparator}{pluginFilePath2}");
+
+            var pluginDiscoverer = new PluginDiscoverer(environmentalVariableReader.Object);
+
+            // Act
+            var plugins = pluginDiscoverer.GetPluginsInNuGetPluginPaths();
+
+            // Assert
+            Assert.Contains(plugins, p => p.Path == pluginFilePath1 && p.RequiresDotnetHost == false);
+            Assert.Contains(plugins, p => p.Path == pluginFilePath2 && p.RequiresDotnetHost == false);
+        }
+
+        [PlatformFact(Platform.Linux)]
+        public void GetPluginsInNuGetPluginPaths_Linux_CaseSensitiveMatching()
```

Comment:
These two tests look almost identical. Rather than having two tests, each of which are platform specific, you have have a single test that uses the BCL's API, or NuGet's API, to check the operating system.